src/test/features/staging-spec.ts

```diff
@@ -116,4 +116,28 @@ suite('ModelViewerElementBase with StagingMixin', () => {
       });
     });
   });
+
+  suite('without a loaded model', () => {
+    setup(async () => {
+      element = new ModelViewerElement();
+      document.body.appendChild(element);
+    });
+
+    teardown(() => {
+      document.body.removeChild(element);
+    });
+
+    suite('auto-rotate', () => {
+      setup(() => {
+        element.autoRotate = true;
+      });
+
+      test('does not cause the model root to rotate over time', async () => {
+        const {turntableRotation} = element;
+        await timePasses(50);  // An arbitrary amount of time, greater than one
+                               // rAF though
+        expect(element.turntableRotation).to.be.equal(turntableRotation);
+      });
```

Comment:
It seems like mixins are intended to no be dependent on each other, so I couldn't specifically test behaviour with having a poster dismissed or not. That does mean that this test also operates under the assumption that if the poster is visible, then the model is not considered "loaded".

Reply:
It's true that we haven't really done tests that integrate multiple mixins. That isn't to say we can't do it. But, I don't expect you to innovate in this regard.

That said, with `poster` and `preload` set together, it's possible for the model to be loaded but still hidden by the poster image.

Reply:
> That said, with `poster` and `preload` set together, it's possible for the model to be loaded but still hidden by the poster image.

That's what I thought, but it doesn't appear to be the case. 

The model is only considered "loaded" with all its children set when `setSource` is called: https://github.com/GoogleWebComponents/model-viewer/blob/master/src/three-components/Model.js#L186

If `poster` is set, then the `setSource` function is still only called once the poster is dismissed: https://github.com/GoogleWebComponents/model-viewer/blob/master/src/features/loading.js#L212. This is regardless of whether `preload` is on it or not.

You can test the behaviour by putting `poster` and `preload` attributes, and putting a breakpoint here https://github.com/GoogleWebComponents/model-viewer/blob/master/src/three-components/Model.js#L186. It will only trigger when you press the poster.

This seems really odd. Conceptually the flow seems weird that a function called `setSource` is called when a poster is dismissed. And I can see that with `preload` the glb is indeed being downloaded, but it is not loaded into the scene hierarchy until the poster is dismissed.

Reply:
You're right, this is a bit confusing.

We have two parallel concepts:

 1. The notion of the model being loaded, which is to say that the model has been loaded over the network and cached by `<model-viewer>`'s internal glTF loader.
 2. The notion of setting the source on the `Model` instance, which causes that class to request the glTF from the loader (which might then load it over the network or pull it from cache) and then subsequently _render_ the model.

When I said it could be loaded, what I meant is that it could have been loaded by the internal glTF loader. We don't invoke `setSource` until interaction in order to avoid rendering before the poster is dismissed, but the model may already be "loaded".

I agree that `setSource` is an odd name given the context in this explanation. You can probably tell by the fact that it is still in JS that this part of the flow dates back to our prototyping days. So, bare with us 🙇

I think the main point is that regardless of whether `setSource` is invoked, the model could be considered "loaded".

Reply:
The model might be "loaded" but `hasModel()` still returns false until the poster is dismissed. Regardless, I prefer the `modelIsVisible` approach.